lib/transformers/allOf.js

```diff
@@ -45,13 +44,72 @@ function generateErrorSummary(transformationOrder, transformations) {
         : '';
 }
 
-function allOf({schema, options, transform, data, schemaErrors} = {}) {
-    const opts = merge({}, options);
-    const groupSchema = schema.allOf[0];
-    const pageHeading = groupSchema.title;
+function schemaContent(schema, nunjucksImports, summaryText, transformations) {
+    const componentsSchema = schema.allOf;
+    function propertyNunjucks(propertySchema) {
+        const {properties} = propertySchema;
+        const propertyIds = Object.keys(properties);
+        return propertyIds
+            .map(propertyId => {
+                const transformation = transformations[propertyId];
+                if (
+                    transformation.componentName === 'rawContent' ||
+                    transformation.componentName === 'summary'
+                ) {
+                    return transformation.content;
+                }
+                return nunjucksHelper.toNunjucks(
+                    transformation.componentName,
+                    transformation.macroOptions
+                );
+            })
+            .join('\n');
+    }
+
+    function beginFieldset(componentSchema) {
+        return `{% from "fieldset/macro.njk" import govukFieldset %}
+        {% call govukFieldset({
+            legend: {
+                html: "${componentSchema.title}",
+                classes: "${
+                    componentsSchema.length === 1
+                        ? `govuk-fieldset__legend--xl`
+                        : `govuk-fieldset__legend--m`
+                }",
+            isPageHeading: ${componentsSchema.length === 1}
+            }
+        }) %}`;
+    }
+
+    function componentContent(componentSchema) {
+        if ('properties' in componentSchema) {
+            return propertyNunjucks(componentSchema);
+        }
+
+        if ('allOf' in componentSchema) {
+            return `${beginFieldset(componentSchema)}
+            ${componentSchema.allOf
+                .map(propertySchema => {
+                    return propertyNunjucks(propertySchema);
+                })
+                .join('\n')}
+        {% endcall %}`;
+        }
+
+        return '';
+    }
+
+    return `${summaryText}${nunjucksImports.join('\n')}
+            ${schema.title ? `<h1 class="govuk-heading-xl">${schema.title}</h1>` : ''}
+            ${componentsSchema
+                .map(componentSchema => {
+                    return `${componentContent(componentSchema)}`;
+                })
+                .join('\n')}`;
+}
 
-    // Transform sub schemas
-    const transformations = groupSchema.allOf.reduce((acc, propertiesSchema) => {
+function transformComponentSchema(componentSchema, options, transform, data, schemaErrors) {
+    function transformPropertySchema(acc, propertiesSchema) {
```

Comment:
can the `transformPropertySchema` function be defined outside and above the `transformComponentSchema` function? or is there a particular reason behind this structure?

Reply:
it's now defined outside.